src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -114,79 +114,98 @@ def create_appliance(resource_group_name,
+    return racf.applications.create_or_update(resource_group_name, application_name, application)
 
 
-def show_appliance(resource_group_name=None, appliance_name=None):
+def show_application(resource_group_name=None, application_name=None):
     """ Gets a managed application.
     :param str resource_group_name:the resource group name
-    :param str appliance_name:the managed application name
+    :param str application_name:the managed application name
     """
     racf = _resource_managedapps_client_factory()
-    return racf.appliances.get(resource_group_name, appliance_name)
+    return racf.applications.get(resource_group_name, application_name)
 
 
-def show_appliancedefinition(resource_group_name=None, appliance_definition_name=None):
+def show_applicationdefinition(resource_group_name=None, application_definition_name=None):
     """ Gets a managed application definition.
     :param str resource_group_name:the resource group name
-    :param str appliance_definition_name:the managed application definition name
+    :param str application_definition_name:the managed application definition name
     """
     racf = _resource_managedapps_client_factory()
-    return racf.appliance_definitions.get(resource_group_name, appliance_definition_name)
+    return racf.application_definitions.get(resource_group_name, application_definition_name)
 
 
-def create_appliancedefinition(resource_group_name,
-                               appliance_definition_name, location,
-                               lock_level, package_file_uri, authorizations,
-                               description, display_name, tags=None):
+def create_applicationdefinition(resource_group_name,
+                                 application_definition_name, location,
+                                 lock_level, authorizations,
+                                 description, display_name,
+                                 package_file_uri=None, create_ui_definition=None,
+                                 main_template=None, tags=None):
     """ Create a new managed application definition.
     :param str resource_group_name:the desired resource group name
-    :param str appliance_definition_name:the managed application definition name
+    :param str application_definition_name:the managed application definition name
     :param str description:the managed application definition description
     :param str display_name:the managed application definition display name
     :param str package_file_uri:the managed application definition package file uri
+    :param str create_ui_definition:the managed application definition create ui definition
+    :param str main_template:the managed application definition main template
     :param str tags:tags in 'a=b c' format
     """
+    if(not package_file_uri and not create_ui_definition and
+       not main_template):
+        raise CLIError('Please specify either --package-file-uri or \
```

Comment:
please use standard usage error format `usage error:  --package-file-uri PACKAGE_FILE_URI | --create-ui-definition DEFINITION  --main-template TEMPLATE`

Reply:
This command accepts only these combinations:
1 - package file uri
2 - create ui def and main template

So standard usage error format won't apply here

Reply:
the `standard` means the error  starts with `usage error`, and also try to use arg-parser style of error reporting